src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringAfterJoin = '';
+    const copyArray = this.slice();
+    let finallySeparator = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (finallySeparator === null) {
+      finallySeparator = 'null';
+    }
+
+    for (let i = 0; i < copyArray.length; i++) {
+      if (copyArray[i] === undefined || copyArray[i] === null) {
```

Comment:
try to invert this condition i think if we go in this direction it might help us to write shorter code
```suggestion
      if (copyArray[i] != null) {
```